sw/device/tests/otbn_randomness_test.c

```diff
@@ -152,6 +154,61 @@ void initialize_clkmgr(void) {
   CLKMGR_TESTUTILS_CHECK_CLOCK_HINT(clkmgr, kOtbnClock, kDifToggleEnabled);
 }
 
+// Reinitialize both EDNs in auto-request (and non-boot) mode.
+static void reinit_edn_auto_mode(void) {
+  const dif_csrng_t csrng = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR)};
+  const dif_edn_t edn0 = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR)};
+  const dif_edn_t edn1 = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_EDN1_BASE_ADDR)};
+
+  // Disable CSRNG.  This is required to abort the generate command of boot
+  // mode, which is likely still running.
+  CHECK_DIF_OK(dif_csrng_stop(&csrng));
+
+  // Stop both EDNs.
+  CHECK_DIF_OK(dif_edn_stop(&edn0));
+  CHECK_DIF_OK(dif_edn_stop(&edn1));
+
+  // Re-enable the CSRNG.
+  CHECK_DIF_OK(dif_csrng_configure(&csrng));
+
+  // Re-enable EDN0 in auto mode.
+  const dif_edn_auto_params_t edn0_params = {
+      // EDN0 provides lower-quality entropy.  Let one generate command return 8
+      // blocks, and reseed every 32 generates.
+      .reseed_material =
+          {
+              .len = 1,
+              .data = {0x00000002},  // reseed from entropy source only
+          },
+      .generate_material =
+          {
+              .len = 1, .data = {0x00008003},  // one generate returns 8 blocks
+          },
+      .reseed_interval = 32,  // reseed every 32 generates
+  };
+  CHECK_DIF_OK(dif_edn_set_auto_mode(&edn0, edn0_params));
+
+  // Re-enable EDN1 in auto mode.
+  const dif_edn_auto_params_t edn1_params = {
+      // EDN1 provides high-quality entropy.  Let one generate command return 1
+      // block, and reseed after every generate.
+      .reseed_material =
+          {
+              .len = 1,
+              .data = {0x00000002},  // reseed from entropy source only
+          },
+      .generate_material =
+          {
+              .len = 1, .data = {0x00001003},  // one generate returns 1 block
+          },
+      .reseed_interval = 1,  // reseed after every generate
+  };
+  CHECK_DIF_OK(dif_edn_set_auto_mode(&edn1, edn1_params));
+}
+
 bool test_main(void) {
   entropy_testutils_boot_mode_init();
```

Comment:
Do we still need this call after these changes?

Reply:
Yes, this function would still call `setup_entropy_src()`, which was required. I dropped the commit that removed the calls to `setup_csrng()` and `setup_edn()` from this function, though, because it fails some FPGA tests. I intend to debug this and create a separate PR.